Releasing 0.6.27 (#3221)
dfinity/motoko: Changelog.md

# Motoko compiler changelog

## 0.6.27 (2022-05-04)

* motoko (`moc`)

>   * Importing modules by relative path is now more robust (#3215).
>   * Performance: externalising `stable var`iables to stable memory is now
>     performed in a streaming manner, reducing heap consumption and
>     copying when an upgrade is being performed (#3149).
>   * Performance: local variables are now stored in unboxed form when
>     possible (thanks to nomeata) (#3207).

Reviewer: ```suggestion
  * Importing modules by relative path is now more robust (#3215).
  * Performance: persisting stable variables to stable memory is now
    performed in streaming fashion, reducing heap consumption and
    copying during an upgrade (#3149).
  * Performance: local 32- and 64-bit numeric values are now stored in unboxed form when
    possible (thanks to nomeata) (#3207).
```

* motoko-base

>   * Fixed a bug in `Trie.filter` (and `Trie.mapFilter`) which could
>     lead to lost matches in case of certain configurations and
>     predicates (dfinity/motoko-base#371).

Reviewer: ```suggestion
  * Fixed a bug in `Trie.filter` (and `Trie.mapFilter`) which could
    lead to missing matches in some cases (dfinity/motoko-base#371).
```
